addons/lunch/static/src/js/lunch_list_renderer.js

```diff
@@ -0,0 +1,59 @@
+odoo.define('lunch.LunchListRenderer', function (require) {
+"use strict";
+
+/**
+ * This file defines the Renderer for the Lunch List view, which is an
+ * override of the ListRenderer.
+ */
+
+var ListRenderer = require('web.ListRenderer');
+
+var LunchListRenderer = ListRenderer.extend({
+    events: _.extend({}, ListRenderer.prototype.events, {
+        'click .o_data_row': '_onClickListRow',
+    }),
+
+    /**
+     * @override
+     */
+    start: function () {
+        this.$el.addClass('o_lunch_view o_lunch_list_view');
+        return this._super.apply(this, arguments);
+    },
+    /**
+     * Override to add id of product_id in dataset.
+     *
+     * @override
+     */
+    _renderRow: function (record) {
+        var tr = this._super.apply(this, arguments);
+        tr.attr('data-product-id', record.data.product_id ? record.data.product_id.res_id : record.data.id);
```

Comment:
I used `record.data.id` because it seems (with the demo data) that `record.data.id` and `record.data.product_id.res_id` are always equals.
Normally `product_id` is always defined.
So I think I can set `data-product-id` to `null` if `product_id` is not defined.